lib/ansible/modules/packaging/os/zypper.py

```diff
@@ -367,7 +382,7 @@ def package_present(m, name, want_latest):
     if any(p.version for p in packages):
         m.params['oldpackage'] = True
 
-    if not want_latest:
+    if not want_latest and not urls:
```

Comment:
?

Reply:
@robinro if i amn't mistaken in this case code tries to find something like this zypper search --match-exact apache-rpm-macros-control-20151110-8.1.x86_64.rpm .  so not sure if zypper search will work in this way

Reply:
`urls` are not checked for presence, but always passed along to zypper. See the rest of this function.